transport/http/client.go

```diff
@@ -85,6 +101,10 @@ func (c Client) Endpoint() endpoint.Endpoint {
 		}
 
 		resp, err := ctxhttp.Do(ctx, c.client, req)
+		for _, f := range c.after {
+			ctx = f(ctx, resp)
+		}
+
```

Comment:
So the reason to have this immediately after the request verses after the err check was to make sure any part of the `http.Response` can be captured, even if an error occurs. To your point above though, for consistency with `Server.ServeHTTP`, it might make sense to have it be mirrored. I'm open either way, what do you think?